common/src/main/java/com/wynntils/core/discord/DiscordManager.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.discord;
+
+import com.wynntils.antiope.core.DiscordGameSDKCore;
+import com.wynntils.antiope.core.type.CreateParams;
+import com.wynntils.antiope.manager.activity.type.Activity;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Manager;
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.TickAlwaysEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.utils.mc.McUtils;
+import java.time.Instant;
+import java.util.List;
+import java.util.Locale;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class DiscordManager extends Manager {
+
+    private DiscordGameSDKCore core;
+    private Activity activity;
+
+    private int level = 0;
+    private ClassType classType = null;
+
+    public DiscordManager() {
+        super(List.of());
+
+        DiscordGameSDKCore.loadLibrary();
+        CreateParams params = new CreateParams();
```

Comment:
Oh, I just noticed. You are leaking memory here with `params`. It is auto closable, but you mustn't close it until you close core. Which means you need to store this as a field too. I am thinking if you should have a holder record for all these so you don't forget to close all of them.